homeassistant/components/pulseaudio_loopback/switch.py

```diff
@@ -58,97 +37,61 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     source_name = config.get(CONF_SOURCE_NAME)
     host = config.get(CONF_HOST)
     port = config.get(CONF_PORT)
-    buffer_size = config.get(CONF_BUFFER_SIZE)
-    tcp_timeout = config.get(CONF_TCP_TIMEOUT)
 
     server_id = str.format("{0}:{1}", host, port)
 
+    if host:
+        connect_to_server = server_id
+    else:
+        connect_to_server = None
+
     if server_id in _PULSEAUDIO_SERVERS:
         server = _PULSEAUDIO_SERVERS[server_id]
     else:
-        server = PAServer(host, port, buffer_size, tcp_timeout)
+        server = Pulse(server=connect_to_server, connect=False, threading_lock=True)
         _PULSEAUDIO_SERVERS[server_id] = server
```

Comment:
Please store the servers in `hass.data` instead. This is legacy code, but would be good to clean it up. Thanks!

Reply:
Done.